src/wp-includes/class-wp-user-query.php

```diff
@@ -362,6 +385,111 @@ public function prepare_query( $query = array() ) {
 			$role__not_in = (array) $qv['role__not_in'];
 		}
 
+		// Capabilities.
+		$available_roles = array();
+
+		if ( isset( $qv['capability'] ) || isset( $qv['capability__in'] ) || isset( $qv['capability__not_in'] ) ) {
```

Comment:
`! empty()` maybe to account for instances in which `fill_query_vars()` is run at the start of `prepare_query()`.